mpas_analysis/test/test_mpas_xarray.py

```diff
@@ -141,11 +140,41 @@ def test_remove_repeated_time_index(self):
                                                              onlyvars=varList,
                                                              yearoffset=1850))
 
-        self.assertEqual(ds.data_vars.keys(), varList)
+        self.assertEqual(sorted(ds.data_vars.keys()), sorted(varList))
         self.assertEqual(len(ds.Time.values), 3)
 
         ds = mpas_xarray.remove_repeated_time_index(ds)
         self.assertEqual(len(ds.Time.values), 2)
 
+    def test_variable_map(self):
+        fileName = str(self.datadir.join('example_jan.nc'))
+        varMap = {
+            'avgSurfaceTemperature':
+                ['time_avg_avgValueWithinOceanRegion_avgSurfaceTemperature',
+                 'other_string',
+                 'yet_another_string'],
+            'daysSinceStartOfSim':
+                ['time_avg_daysSinceStartOfSim',
+                 'xtime',
+                 'something_else'],
+            'avgLayerTemperature':
+                ['time_avg_avgValueWithinOceanLayerRegion_avgLayerTemperature',
+                 'test1',
+                 'test2'],
+            'Time': [['xtime_start', 'xtime_end'],
+                     'time_avg_daysSinceStartOfSim']}
+
+        varList = ['avgSurfaceTemperature', 'avgLayerTemperature',
+                   'refBottomDepth', 'daysSinceStartOfSim']
+
+        ds = xr.open_mfdataset(
+            fileName,
+            preprocess=lambda x: mpas_xarray.preprocess_mpas(
+                x,
+                timestr='Time',
+                onlyvars=varList,
+                yearoffset=1850,
+                variable_map=varMap))
+        self.assertEqual(sorted(ds.data_vars.keys()), sorted(varList))
```

Comment:
I'd put a comment here that the preprocess function is essentially remapping variables within the dataset, otherwise the assert statement could be a little confusing.